solution.cpp

```diff
@@ -10,125 +10,131 @@
- *   be confusing.
- */
-void sieve(const int size, bool primes[]);
+#include <algorithm>
+using std::sort;
+using std::find;
+using std::begin;
+using std::end;
+using std::remove;
 
-/**
- * Read numbers from `infilename`, and if they are prime, output them to
- * `outfilename`, separated by newlines.
- *
- * Arguments:
- * - `size`: The size of (the number of elements in) the array.
- * - `primes`: An array of boolean values, with each element set to `true` if
- *   its index is prime, and `false` otherwise.
- * - `infilename`: A string representing the file path to read from.
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_primes( const int size,
-                  bool primes[],
-                  const string& infilename,
-                  const string& outfilename );
+//function to show off all the hard work to the screen
+void displayToScreen(vector<int>, vector<int>);
 
-/**
- * Read numbers from `infilename`, and if they are composite, output them to
- * `outfilename`, separated by newlines.
- *
- * Arguments:
- * - `size`: The size of (the number of elements in) the array.
- * - `primes`: An array of boolean values, with each element set to `true` if
- *   its index is prime, and `false` otherwise.
- * - `infilename`: A string representing the file path to read from.
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_composites( const int size,
-                      bool primes[],
-                      const string& infilename,
-                      const string& outfilename );
-
-// ----------------------------------------------------------------------------
-
-int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
-        return 1;  // error
-    }
-
-    bool primes[max+1];
-    sieve(max+1, primes);
-
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
-    }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
-    }
-
-    return 0;  // success
+
+int main()
+{
+
+	//open a file
+	ifstream input("input.txt");
+
+
+	//ensure that file is open and good to go
+	if (!input.is_open()){
+		cout << "Can't find input.txt file" << endl;
+		return 1;
+	}
+
+	//Declare some variables as well as vectors//
+
+	//Used to temporarily hold value fetched from file
+	int number;
+	
+
+	vector<int> primes;
+	vector<int> composites;
+	
+	cout << "List of numbers from file: ";
+
+	//read values into array
```

Comment:
you shouldn't need to read in and store all the numbers at once -- you should be able to process them as they're read in